clang/lib/AST/PreorderAST.cpp

```diff
@@ -404,6 +404,12 @@ bool BinaryOperatorNode::ConstantFold(bool &Error, ASTContext &Ctx) {
       ConstFoldedVal = CurrConstVal;
 
     } else {
+      // Ensure that ConstFoldedVal and CurrConstVal have the same bit width.
+      if (ConstFoldedVal.getBitWidth() < CurrConstVal.getBitWidth())
+        ConstFoldedVal = ConstFoldedVal.extOrTrunc(CurrConstVal.getBitWidth());
+      if (CurrConstVal.getBitWidth() < ConstFoldedVal.getBitWidth())
```

Comment:
This should be an `else if` because either `ConstFoldedVal.getBitWidth() < CurrConstVal.getBitWidth()` or `CurrConstVal.getBitWidth() < ConstFoldedVal.getBitWidth()`, but not both.